app/views/rooms/show.html.erb

```diff
@@ -83,13 +83,63 @@ $(function(){
       var $ol = $("div.reviews ol")
       $ol.html("") // empties the ol
       json.reviews.forEach(function(review){
-        $ol.append("<li>" + review.content + "</li>")
+        if (review.user_id === <%= current_user.id %>){
+          $ol.append(
+            `<li>
+              ${review.content}
+              <button
+                class="delete_review"
+                data-review-id="${review.id}"
+                data-room-id="${review.room_id}"
+              >
+                Delete
+              </button>
+            </li>`
+          )
+        } else {
+          $ol.append(`<li>${review.content}</li>`)
+        }
+        bindDeleteReviewEventListeners();
       })
     })
 
     e.preventDefault();
   })
 
+  function bindDeleteReviewEventListeners() {
+    $("button.delete_review").on("click", function(e){
+      $.ajax({
+        type: "DELETE",
+        url: `/rooms/${this.dataset.roomId}/reviews/${this.dataset.reviewId}`,
+        success: function(json){
```

Comment:
This success should not be returning JSON. You should only have to remove the e.target from the DOM and unbind the delete event for that element instead of making a new list of reviews with delete buttons.